packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -227,6 +227,13 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
       if (!torrent.done && wire.amChoking) {
         wire._clearTimeout();
       }
+
+      if (!wire.amChoking && wire.downloaded === wire['prevDownloaded']) {
+        this.closeChannel(wire, wire.paidStreamingExtension.leechingChannelId);
+      } else {
+        wire['prevDownloaded'] = wire.downloaded;
+      }
```

Comment:
Not sure if this is the best way to retain this tiny bit of local data on the wire but it's convenient and short.

Reply:
To be consistent, I would put this data in the paidStreamingExtension. All of web3torrent's wires extended data that we handle is in there already. Seems like a cleaner solution than this.